### ISSUE TITLE
We should respect the numerical tower (PEP 3141) for standard numeric types

### ISSUE THREAD

**Issue by gvanrossum** [ambv/typehinting#75]
Even though this isn't the case for the implementation, we should consider int a subclass of float, and float a subclass of complex. The alternative is that we'd have to sprinkle things like this all over code, and this is just too common to bother:
```
from numbers import Float, Integer
def foo(x: Integer, y: Integer) -> Float: ...
```
However, we shouldn't special-case Decimal: it's too quirky, and even numeric.py states "Decimal [...] should not be registered as a Real because decimals do not interoperate with binary floats".

**Comment by JukkaL** [comment]
It might be somewhat reasonable to treat `int` as a subtype of `Decimal`, since these types are mostly interoperable. However, `int` doesn't have all the methods supported by `Decimal`, so this could actually mask runtime type errors, and requiring explicit `int` -> `Decimal` conversions or using `Union[int, Decimal]` is probably a better thing to do.

I agree that `Decimal` should be distinct from `float`.

**Comment by gvanrossum** [comment]
Closed by rev f6c17dd8ccca955374e465172de44e04b0a6c037.
